# HG changeset patch
# User Torsten <mttl@mailbox.org>
# Date 1522784933 -7200
#      Tue Apr 03 21:48:53 2018 +0200
# Branch stable
# Node ID 65b7380655f2c7e4f1e66003627e5cf09e918f3a
# Parent  29641449ca90bd3dcc34287ea5b6c849f78a7847
prevent crash if help data could not be set up (bug #53006)

* documentation.cc (documentation): return before creating documentation
  widget if help data failed to be set up;
  (~documentation): remove tmp files even if help engine is null pointer

diff --git a/libgui/src/documentation.cc b/libgui/src/documentation.cc
--- a/libgui/src/documentation.cc
+++ b/libgui/src/documentation.cc
@@ -82,20 +82,19 @@ namespace octave
 
     if (! m_help_engine->setupData())
       {
         QMessageBox::warning (this, tr ("Octave Documentation"),
                               tr ("Could not setup the data required for the\n"
                                   "documentation viewer. Only help texts in\n"
                                   "the Console Widget will be available."));
         if (m_help_engine)
-          {
-            delete m_help_engine;
-            m_help_engine = 0;
-          }
+          delete m_help_engine;
+        m_help_engine = 0;
+        return;
       }
 
     // The browser
     m_doc_browser = new documentation_browser (m_help_engine, this);
 
     // Layout contents, index and search
     QTabWidget *navi = new QTabWidget (this);
     navi->setTabsClosable (false);
@@ -183,21 +182,22 @@ namespace octave
     // Initial view: Contents
     m_doc_browser->setSource (QUrl (
         "qthelp://org.octave.interpreter-1.0/doc/octave.html/index.html"));
   }
 
   documentation::~documentation (void)
   {
     if (m_help_engine)
-      {
-        delete m_help_engine;
+      delete m_help_engine;
 
-        // Cleanup temporary file and directory
-        QFile file (m_collection);
+    // Cleanup temporary file and directory
+    QFile file (m_collection);
+    if (file.exists ())
+      {
         QFileInfo finfo (file);
         QString bname = finfo.fileName ();
         QDir dir = finfo.absoluteDir ();
         dir.setFilter (QDir::Dirs | QDir::NoDotAndDotDot | QDir::Hidden);
         QStringList namefilter;
         namefilter.append ("*" + bname + "*");
         foreach (QFileInfo fi, dir.entryInfoList (namefilter))
           {
